bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/Expression.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.TimeZone;
+
+/**
+ * <code>Expression<code> is the interface that all Expressions have to implement. Expression are typically evaluated as
+ * from a given start date in a given timezone. Once set, the expression can be queried for the next date, from the
+ * start date, that will match the expression
+ *
+ * @author Karel Goderis
+ *
+ */
+public interface Expression {
```

Comment:
Yes! The start date is the date from which you want the expresion to be evaluated. The start date is not perse encoded in that expression, so it should/could be set afterwards. The same is true for the timezone, as some expressions do not capture timezone information.